src/modules/quick_molspec_module.f90

```diff
@@ -193,25 +193,34 @@ subroutine reallocate_quick_molspec(self,ierr)
      integer, intent(inout) :: ierr
      integer :: current_size
 
-     if(self%nextatom .gt. 0) then
-       if(allocated(self%extchg)) current_size= size(self%extchg)
-       if(current_size /= self%nextatom) then
+     if(self%nextatom > 0) then
+       if(allocated(self%extchg)) then
+         current_size = size(self%extchg)
+       else
+         current_size = 0
+       endif
+       if(current_size < self%nextatom) then
          deallocate(self%extchg, stat=ierr)
          deallocate(self%extxyz, stat=ierr)
          allocate(self%extchg(self%nextatom), stat=ierr)
          allocate(self%extxyz(3,self%nextatom), stat=ierr)
-         self%extchg=0.0d0
-         self%extxyz=0.0d0
        endif
+       self%extchg=0.0d0
+       self%extxyz=0.0d0
      endif
 
-     if(self%nextpoint .gt. 0) then
-      if(current_size /= self%nextpoint) then
+     if(self%nextpoint > 0) then
+       if(allocated(self%extpointxyz)) then
+         current_size = size(self%extpointxyz)
+       else
+         current_size = 0
+       endif
+      if(current_size < 3*self%nextpoint) then
```

Comment:
Need to check that dimension of these arrays can be larger than required where these arrays are used.